backend/space/validators.py

```diff
@@ -0,0 +1,15 @@
+from django.core.exceptions import ValidationError
+from django.db.models import Choices, IntegerChoices
+
+
+def get_label(value, options: Choices):
+    return next(label for (v, label) in options.choices if v == value)
+
+
+def validate_level_deeper_than_parent(level, parent, options: IntegerChoices):
+    if not level > parent.level:
```

Comment:
Should we allow for structures of one level to be a child of structures of the _same_ level? I tried to add the 'Poitiers nunnery' (part of a monastery, but multiple rooms, so level 3: building) with 'The Poitiers monastery' (level 3) as its parent, but that was not allowed.

I am not sure how real this is, though, so we could also simply put this as a question to Robert/Hope.

Reply:
Yes, I thought we might need that, but let's see what the researchers think.

We could also adjust this function so structures of _some_ levels (e.g. buildings) accept children of the same level, while others don't.

Though in the current hierarchy, I think you could add your example as:

The Poitiers Monastery (Fortification) < Nunnery (Building) < A cell (Room)